lib/ask/runtime/broker.ex

```diff
@@ -86,10 +86,10 @@ defmodule Ask.Runtime.Broker do
       reached_quotas = reached_quotas?(survey)
 
       cond do
-        reached_quotas || (active == 0 && ((pending + stalled) == 0 || survey.cutoff <= completed)) ->
+        (active == 0 && ((pending + stalled) == 0 || survey.cutoff <= completed || reached_quotas)) ->
           complete(survey)
 
-        active < batch_size() && pending > 0 ->
+        active < batch_size() && pending > 0 && !(survey.cutoff <= completed || reached_quotas) ->
```

Comment:
@lmatayoshi @nekron Maybe extract this condition to a variable? 

```
survey.cutoff <= completed || reached_quotas
```

With so many `&&` and `||` it's hard to follow the logic.

(before this change it was also like that, but we should improve the readability of this chunk of code)